### DISCUSSION TITLE
Include docs sidebar on 404 page?

### DISCUSSION THREAD

**Discussion by Nirusu** [facebook/docusaurus#8089] [Q&A]
I would like to include the sidebar from the docs pages to the 404 page so users who might have mistyped an URL can easily get to the contents in the docs without going to the root page first.

I found that you can [swizzle](https://github.com/facebook/docusaurus/discussions/6030) the 404 page, but I am not sure how I could include a sidebar here. Looking at the code from other pages (such as `DocPage`), the sidebar data is retrieved over `useDocRouteMetadata`, which requires `props`. However, this is not passed to NotFound, so I don't think I can import the sidebar like this.

Is it be possible to get the sidebar included in the NotFound page via swizzling? If yes, how can I achieve this?

**Comment by Josh-Cena** [comment]
FYI, Sebastien and I had a discussion on this:

![image](https://user-images.githubusercontent.com/55398995/189710093-b7978e86-d15c-4962-88e4-0fa0e1c400d7.png)

![image](https://user-images.githubusercontent.com/55398995/189710106-a53bdb39-58cc-4222-ad3e-b855fe481b75.png)

![image](https://user-images.githubusercontent.com/55398995/189710118-340d68b8-d862-4545-826e-25faa0f62a88.png)

You would need to also swizzle `NotFound` and make it display the docs sidebar—otherwise the initial 404.html will be different from the final content.

**Reply by Nirusu** [reply]
Mh, I figured that it requires swizzling `NotFound`... but I am not sure how to achieve this. 
I am not really familiar with React, so I might be missing something... Is it somewhat trivial? 

Judging from your Discord conversation, it sounds like it is a bit more complex endeavor. If it is I might just wait for a future release, but if you have any pointers on how to achieve this with swizzling I would really appreciate it :) 

It doesn't need to be super dynamic, my setup only has one docs sidebar only anyways so if I can get that working that would already be fine.

**Reply by slorber** [reply]
It is indeed not so trivial even for us maintainers to implement in our own site, and even more to create a generalized solution that is easy to apply for our users

I suggest you swizzle NotFound and just display a hardcoded list of popular docs that you recommend the user to check, instead of trying to get the real docs sidebar display there.
